src/eigenlayer/contracts.py

```diff
@@ -0,0 +1,192 @@
+    ) -> int:
+        withdrawals = await self.contract.functions.getClaimableUserDelayedWithdrawals(
+            address
+        ).call(block_identifier=block_number)
+        return len(withdrawals)
+
+
+class EigenPodManagerContract(ContractWrapper):
+    abi_path = '../eigenlayer/abi/IEigenPodManager.json'
+    settings_key = 'EIGEN_POD_MANAGER_CONTRACT_ADDRESS'
+
+    async def get_last_pod_shares_updated_event(
+        self, pod_owner: ChecksumAddress, from_block: BlockNumber, to_block: BlockNumber
+    ) -> EventData:
+        return await self._get_last_event(
+            event=self.events.PodSharesUpdated,  # type: ignore
+            from_block=from_block,
+            to_block=to_block,
+            argument_filters={'podOwner': pod_owner},
+        )
+
+    async def get_beacon_chain_oracle(self, block_number: BlockNumber) -> ChecksumAddress:
+        address = await self.contract.functions.beaconChainOracle().call(
+            block_identifier=block_number
+        )
+        return Web3.to_checksum_address(address)
+
+    async def get_pod_shares(self, pod_owner: ChecksumAddress, block_number: BlockNumber) -> Wei:
+        shares = await self.contract.functions.podOwnerShares(pod_owner).call(
+            block_identifier=block_number
+        )
+        return Wei(shares)
+
+
+class BeaconChainOracleContract(ContractWrapper):
+    abi_path = '../eigenlayer/abi/IBeaconChainOracle.json'
+
+    def __init__(self, address):
+        self._address = address
+
+    @property
+    def contract_address(self) -> ChecksumAddress:
+        return self._address
+
+    async def get_last_oracle_update_event(
+        self, from_block: BlockNumber, to_block: BlockNumber
+    ) -> EventData:
+        return await self._get_last_event(
+            event=self.events.EigenLayerBeaconOracleUpdate,  # type: ignore
+            from_block=from_block,
+            to_block=to_block,
+        )
+
+
+class DelegationManagerContract(ContractWrapper):
+    abi_path = '../eigenlayer/abi/IDelegationManager.json'
+    settings_key = 'EIGEN_DELEGATION_MANAGER_CONTRACT_ADDRESS'
+
+    async def get_withdrawal_queued_events(
+        self, from_block: BlockNumber, to_block: BlockNumber
+    ) -> list[QueuedWithdrawal]:
+        events = await self._get_events(
+            event=self.events.WithdrawalQueued,  # type: ignore
+            from_block=from_block,
+            to_block=to_block,
+        )
+        return [QueuedWithdrawal(*event['args']['withdrawal'].values()) for event in events]
+
+    async def get_withdrawal_completed_events(
+        self, from_block: BlockNumber, to_block: BlockNumber
+    ) -> list[QueuedWithdrawal]:
+        events = await self._get_events(
+            event=self.events.WithdrawalCompleted,  # type: ignore
+            from_block=from_block,
+            to_block=to_block,
+        )
+        return [QueuedWithdrawal(*event['args']['withdrawalRoot']) for event in events]
```

Comment:
`event['args']['withdrawalRoot']` is bytes32 as I see for WithdrawalCompleted event. You can not simply wrap in QueuedWithdrawal.